sources/packages/api/src/services/student-account-applications/student-account-applications.service.ts

```diff
@@ -0,0 +1,197 @@
+      },
+    });
+  }
+
+  /**
+   * Create a new student account application to be reviewed by
+   * the Ministry to confirm the student's basic BCeID identity.
+   * @param userName user name that uniquely identifies this user.
+   * @param studentProfile information to be assessed by the Ministry.
+   * @returns student account application created id.
+   */
+  async createStudentAccountApplication(
+    userName: string,
+    studentProfile: StudentAccountApplicationCreateModel,
+  ): Promise<StudentAccountApplication> {
+    const newUser = new User();
+    newUser.userName = userName;
+    newUser.email = studentProfile.email;
+    newUser.firstName = studentProfile.firstName;
+    newUser.lastName = studentProfile.lastName;
+    newUser.creator = newUser;
+
+    const newAccountApplication = new StudentAccountApplication();
+    newAccountApplication.user = newUser;
+    newAccountApplication.creator = newUser;
+    newAccountApplication.submittedData = studentProfile;
+    newAccountApplication.submittedDate = new Date();
+    return this.repo.save(newAccountApplication);
+  }
+
+  /**
+   * Approve the student account application associating the user
+   * with a student account. The Ministry can also adjust any student
+   * data that will then be used to create the student account.
+   * @param id student account application id.
+   * @param studentProfile user and student data to be updated.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   */
+  async approveStudentAccountApplication(
+    id: number,
+    studentProfile: StudentAccountApplicationApprovalModel,
+    auditUserId: number,
+  ) {
+    const accountApplication = await this.repo.findOne({
+      select: { id: true, user: { id: true } },
+      relations: { user: true },
+      where: { id },
+    });
+
+    if (!accountApplication) {
+      throw new CustomNamedError(
+        "Student account application not found.",
+        STUDENT_ACCOUNT_APPLICATION_NOT_FOUND,
+      );
+    }
+
+    const userInfo = {
+      userId: accountApplication.user.id,
+      lastName: studentProfile.lastName,
+      givenNames: studentProfile.firstName,
+      email: studentProfile.email,
+      birthdate: getISODateOnlyString(studentProfile.dateOfBirth),
+      gender: studentProfile.gender,
+    } as UserInfo;
+
+    const studentInfo = {
+      phone: studentProfile.phone,
+      sinNumber: studentProfile.sinNumber,
+      addressLine1: studentProfile.addressLine1,
+      addressLine2: studentProfile.addressLine2,
+      provinceState: studentProfile.provinceState,
+      country: studentProfile.country,
+      city: studentProfile.city,
+      postalCode: studentProfile.postalCode,
+      selectedCountry: studentProfile.selectedCountry,
+    };
+
+    return this.dataSource.transaction((entityManager) => {
+      // Update the student account application with the approval.
+      const auditUser = { id: auditUserId } as User;
```

Comment:
Just a doubt, will all the promises be resolved when we await for the service method which will in turn await for datasource.transaction ? I'm doubtful on line 143 but a transaction must resolve all the promises together.

Reply:
There is a missing `await` at 143.